lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -2259,9 +2261,11 @@ static bool collectInnermostTensorFlowDTypes(
 /// This deletes the underlying inst in `opInfo` when a GraphOperation is
 /// created successfully.
 void TFDeabstraction::evaluateAttributesAndDoPacking(
-    GraphOperationInfo &opInfo, DenseMap<SILValue, SymbolicValue> &constants,
+    GraphOperationInst *origInst, GraphOperationInfo &opInfo,
+    DenseMap<SILValue, SymbolicValue> &constants,
     GraphFunctionDeviceInfo &deviceInfo) {
-  auto *origInst = opInfo.getInst();
+  assert(opInfo.getInst() == origInst &&
```

Comment:
given this invariant, not passing `origInst` into the func call would seem to reduce the interface complexity. WDYT?

Reply:
GraphOperationInfo::getInst() returns a const pointer, but this function needs to have a non-const pointer to make changes to the instruction. We will need to pass origInst, but can skip passing the GraphOperationInfo with a little bit of refactoring. I was planning to do that anyways.